core/src/Kokkos_Core.hpp

```diff
@@ -316,7 +332,7 @@ std::vector<ExecSpace> partition_space(ExecSpace const& space,
       "Kokkos Error: partitioning arguments must be integers or floats");
 
   std::vector<ExecSpace> instances(weights.size());
```

Comment:
I would have created an empty vector and reserved storage.
https://github.com/kokkos/kokkos/pull/8114#discussion_r2121174573
I am not blocking but you will need approval from someone else.